src/main/java/com/bobocode/petros/container/DependencyDefinition.java

```diff
@@ -1,20 +1,35 @@
 package com.bobocode.petros.container;
 
+import com.bobocode.petros.exception.NoDefaultConstructorException;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashSet;
 import java.util.Set;
 
-@SuppressWarnings("all")
+@Slf4j
+@ToString
 public class DependencyDefinition {
     private String name;
     private String qualifiedName;
     private Set<DependencyDefinition> injectedDependencyDefinitions;
     private static Object dependency;
 
+    public DependencyDefinition() {
+        injectedDependencyDefinitions = new HashSet();
+    }
+
     @SuppressWarnings("unchecked")
-    public <T> T getDependencyClass() throws Exception {
+    public Object getDependencyClass() {
         if (dependency == null) {
-            dependency = (T) Class.forName(qualifiedName).getConstructor().newInstance();
+            try {
+                dependency = Class.forName(qualifiedName).getConstructor().newInstance();
+            } catch (Exception e) {
+                LOG.error("No default constructor found for class {}. Please, declare default constructor", qualifiedName);
```

Comment:
I think a request like "Please do something" is not very suitable for logging, it is ok for compile-time exception messages, but I am not sure it has some sense in an error log. 
It is not a strong opinion, just an observation =)

Reply:
We're using logging to communicate with a user. I agree that it's not the right place for that log. I was thinking to use Error logs also with hints how to fix it at exception handling level. For now I suggest to left this log as it is and after we will work on exception handling - make full revision of our logging and move such logs on exception handling level

Reply:
Done